More data

zmoon opened the pull request.

zmoon committed:
Initial soil moisture attempt

zmoon committed:
10-m wind from sfc

zmoon committed:
Downward shortwave at sfc

zmoon committed:
More levels

zmoon committed:
Relative

zmoon committed:
New variables; up to 10 km

zmoon committed:
Read new variables

zmoon committed:
Merge branch 'plumerise' into more-data

zmoon committed:
Chem_Allocate?

zmoon committed:
Add plume rise lib for API

zmoon commented:
@bbakernoaa I had to modify some things about the plume rise build config to get it to build.

bbakernoaa commented:
Oops yes. I have some local changes too


Barry Baker

National Oceanic and Atmospheric Administration
Air Resources Laboratory
Physical Research Scientist
Chemical Modeling and Emissions Group Leader
NCWCP, R/ARL, Rm. 4204
5830 University Research Court
College Park, Maryland  20740
Phone: ‪(301) 683-1395‬



On Mon, Jul 29, 2024 at 7:31 PM Zachary Moon ***@***.***>
wrote:

> @bbakernoaa <https://github.com/bbakernoaa> I had to modify some things
> about the plume rise build config to get it to build.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bbakernoaa/CATChem/pull/35#issuecomment-2257181167>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFIUVNZ2PC65U5DAHXJIPILZO3GGFAVCNFSM6AAAAABLVF5QA2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDENJXGE4DCMJWG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

zmoon committed:
Merge branch 'plumerise' into more-data

zmoon committed:
Not PAR

zmoon committed:
Match MetState name

zmoon committed:
Match MetState name and convert

zmoon committed:
More adjustments to variables

- alpha order in the code
- (group, alpha) order in the YAML
- some unit conversions
- keys in YAML same as MetState members
- some things removed that seem not to match MetState

zmoon committed:
Towards running multiple cases

zmoon committed:
Fix null nz

zmoon committed:
Fix wilting point src

zmoon committed:
Regen profiles

zmoon committed:
Convert landuse type codes to int

zmoon committed:
Regen columns

zmoon committed:
Advance line when variable not used

zmoon committed:
Read SNODP normally

zmoon committed:
FRVEG should be float

zmoon committed:
Towards reading code gen

zmoon committed:
Allow skipping reading some vars

zmoon committed:
Fixes

zmoon committed:
Some fixes, including printing loaded stuff

zmoon committed:
Include size

zmoon committed:
Use size fn

zmoon committed:
Regen loader

bbakernoaa commented on util/get-vars.py:
```diff
@@ -2,112 +2,219 @@
+        ds["gwetroot"] = (ds["soilw2"] * 30 + ds["soilw3"] * 60) / 90
+
+        # Combine soil moisture into one array
+        ds["soilm"] = xr.concat([ds[f"soilw{i}"] for i in range(1, 5)], dim="soil")
```
note we should make this flexible - was just in a meeting today for SFS and they are talking about expanding the number of soil levels (perhaps to 9)

zmoon replied:
Sure I can change it now to search for "soilw" variables instead of hardcoding the numbers. But a soil dimension in the dataset would be better if they can do that.

zmoon replied:
Addressed in 7f7208d

zmoon commented on util/get-vars.py:
```diff
@@ -2,112 +2,219 @@
+        das.append(da.rename(vn))
+
+    # Write to text file
+    # TODO: if the number of levels is too large it currently writes multiple lines (truncation to next line) in the output
+    #       Needs to be on a single line
```
```suggestion
```
@bbakernoaa I didn't encounter this issue, maybe you can try it again on yours

zmoon commented on util/vars.yml:
```diff
@@ -2,92 +2,112 @@
-  atmFile: 'atmf012.nc'
-  sfcFile: 'sfcf012.nc'
-  srcLoc: "/scratch1/NCEPDEV/stmp2/Barry.Baker/FV3_RT/rt_980816/atmaero_control_p8_intel"
+cases:
```
@bbakernoaa can you check these are consistent with what you wanted?

bbakernoaa replied:
looks good

zmoon committed:
Auto soil moistures

bbakernoaa committed:
Update util/get-vars.py

Co-authored-by: Zachary Moon <zmoon92@gmail.com>
